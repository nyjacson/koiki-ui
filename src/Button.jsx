import React, { Component, PropTypes } from 'react';
import fa from './less/fa/less/font-awesome.less';

// eslint-disable-next-line react/prefer-stateless-function
class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {
      clicked: false,
      escaped: false
    };
  }

  render() {
    return (
      <div
        className={`${this.props.styles.container} ${this.state.clicked ? this.props.styles.clicked :
                                                     this.state.escaped ? this.props.styles.escaped : ''}`}
      >
        <button
          className={`${this.props.styles.button} ${this.props.disabled ? this.props.styles.disabled : ''}`}
          onClick={() => {
            if (!this.props.disabled) {
              this.setState({
                clicked: this.state.clicked !== true,
                escaped: this.state.clicked === true
              });
              this.props.onClick();
            }
          }}
        >
          <i className={`${fa.fa} ${fa[this.props.icon]}`} aria-hidden="true" />{this.props.text}
        </button>
      </div>
    );
  }
}

Button.propTypes = {
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  styles: PropTypes.object
};

Button.defaultProps = {
  styles: require('./less/button.less'),
  disabled: false,
  icon: 'fa-search',
  onClick: () => {}
};

export default Button;
