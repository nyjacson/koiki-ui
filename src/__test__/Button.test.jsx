import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button text="Search" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
