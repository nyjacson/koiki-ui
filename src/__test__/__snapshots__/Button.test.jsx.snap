exports[`<Button /> renders correctly 1`] = `
<div
  className="undefined ">
  <button
    className="undefined "
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="undefined undefined" />
    Search
  </button>
</div>
`;
